packages/client-sdk-nodejs/src/preview-leaderboard-client.ts

```diff
@@ -0,0 +1,35 @@
+import {MomentoLogger} from '@gomomento/sdk-core';
+import {LeaderboardDataClient} from './internal/leaderboard-client';
+import {LeaderboardClientProps} from './leaderboard-client-props';
+import {ILeaderboardClient} from '@gomomento/sdk-core/dist/src/clients/ILeaderboardClient';
+import {InternalLeaderboardClient} from '@gomomento/sdk-core/dist/src/internal/clients';
+
+/**
+ * PREVIEW Momento Leaderboard Client
+ * WARNING: the API for this client is not yet stable and may change without notice.
+ * Please contact Momento if you would like to try this preview.
+ *
+ * Leaderboard methods return a response object unique to each request.
+ * The response object is resolved to a type-safe object of one of several
+ * sub-types. See the documentation for each response type for details.
+ */
+export class PreviewLeaderboardClient implements ILeaderboardClient {
+  protected readonly logger: MomentoLogger;
+  protected readonly props: LeaderboardClientProps;
+
+  constructor(props: LeaderboardClientProps) {
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    this.logger.debug('Creating Momento LeaderboardClient');
+    this.props = props;
+  }
+
+  /**
+   * Creates an instance of LeaderboardClient with 32-bit float scores.
+   */
+  public createLeaderboard(
+    cacheName: string,
+    leaderboardName: string
+  ): InternalLeaderboardClient {
```

Comment:
This class and this function are public; that means that their return types also need to be public. So you will want to create a public interface for this return type, most likely called `ILeaderboard`.

If you want to make it so that `InternalLeaderboardClient` implements that interface for now, that's fine. But that's a private type, so we shouldn't leak it across the public API boundary.

Also, one thing to keep in mind for the future - doesn't have to be addressed before bug bash or before we share with Jah - but we are currently creating the gRPC channel in the LeaderboardDataClient. That means that with your current implementation, we will create a new gRPC channel for each `leaderboard`, which probably isn't necessary. It would probably be more efficient to keep the internal leaderboard client implementation more closely aligned with the proto-generated code, where all of its signatures accept the cache name and the leaderboard name. Then in this class you would only keep one instance of that internal client around.

In that model you would have a (possibly private/internal) class called `Leaderboard` or `MomentoLeaderboard`, which would provide a concrete implementation of `ILeaderboard`. When you called its constructor you would pass in the cache name and leaderboard name, as well as the internal LeaderboardDataClient, and it would save a reference to each of them. That way we are able to share the single gRPC connection across multiple instances of `ILeaderboard`.

Let's get ready for the bug bash first - so as long as you make the public-facing API changes for now, we are good - but maybe consider a fast follow to tweak this for connection management as soon as we think the API has settled.